open-craft/opencraft
# BB-1714 Prevent Consul collision on integration tests

Intro
===

This PR adds a random ID to the OpenEdxInstance created during integration tests. This is necessary as the instance ID is used as a prefix to Consul data and multiple instances are created in parallel.

Testing
---

1. Checkout this branch and start the devstack.
2. Run the steps used by CircleCI to run Consul locally:
```bash
sudo wget -P /tmp https://releases.hashicorp.com/consul/1.2.1/consul_1.2.1_linux_amd64.zip
sudo unzip /tmp/consul_1.2.1_linux_amd64.zip -d /usr/local/bin
consul agent -dev
```
3. Fire up the shell and use the modified factory method to create an instance:
```python
from instance.tests.integration.factories.instance import OpenEdXInstanceFactory                                                
i = OpenEdXInstanceFactory()
print(i.id)
```
4. Check the Consul log for the prefix containing the random ID.
```
    2019/09/26 21:09:00 [DEBUG] http: Request GET /v1/kv/ocim/instances/1488912035 (106.941µs) from=127.0.0.1:42116
```
5. Stop the Consul process and try to get an id, making sure it works:
```
from instance.tests.integration.factories import instance
print(instance._get_unused_id())
```
